packages/@react-spectrum/s2/src/Tabs.tsx

```diff
@@ -10,9 +10,302 @@
+          [IconContext, {styles: icon}]
+        ]}>
+        {props.children}
+      </Provider>
+    </RACTab>
+  );
+}
+
+const tablist = style({
+  display: 'flex',
+  height: {
+    orientation: {
+      horizontal: {
+        density: {
+          compact: 32,
+          regular: 48
+        }
+      }
+    }
+  },
+  gap: {
+    orientation: {
+      horizontal: {
+        density: {
+          compact: 24,
+          regular: 32
+        }
+      }
+    }
+  },
+  flexDirection: {
+    orientation: {
+      vertical: 'column'
+    }
+  },
+  paddingEnd: {
+    orientation: {
+      vertical: 20
+    }
+  },
+  paddingStart: {
+    orientation: {
+      vertical: 12
+    }
+  },
+});
+
+export function TabList<T extends object>(props: TabListProps<T>) {
+  let {density, isDisabled, disabledKeys, orientation = 'horizontal'} = useContext(TabsInternalContext);
+  let state = useContext(TabListStateContext);
+  let [selectedTab, setSelectedTab] = useState<HTMLElement | undefined>(undefined);
+  let tablistRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {    
+    if (tablistRef?.current) {
+      let tab: HTMLElement | null = tablistRef.current.querySelector('[role=tab][data-selected=true]');
+
+      if (tab != null) {
+        setSelectedTab(tab);
+      }
+    }
+
+  }, [tablistRef, state?.selectedItem]);
+
+  return (
+    <div className={style({position: 'relative'})}>
+      {orientation === 'vertical' && 
+        <TabLine disabledKeys={disabledKeys} isDisabled={isDisabled} selectedTab={selectedTab} orientation={orientation} />}
+      <RACTabList 
+        {...props}
+        ref={tablistRef}
+        className={renderProps => tablist({...renderProps, density})} />
+      {orientation === 'horizontal' && 
+        <TabLine disabledKeys={disabledKeys} isDisabled={isDisabled} selectedTab={selectedTab} orientation={orientation} />}
+    </div>
+  );
+}
+
+function isAllTabsDisabled<T>(collection: Collection<Node<T>> | null, disabledKeys: Set<Key>) {
+  let selectedKey: Key | null = null;
+  if (collection) {
+    selectedKey = collection.getFirstKey();
+    
+    let index = 0;
+    while (selectedKey && index < collection.size) {
+      // We have to check if the item in the collection has a key in disabledKeys or has the isDisabled prop set directly on it
+      if (!disabledKeys.has(selectedKey) && !collection.getItem(selectedKey)?.props?.isDisabled) {
+        return false;
+      }
+
+      selectedKey = collection.getKeyAfter(selectedKey)
+      index++;
+    }
+    return true;
+  }
+  return false;
+}
+
+interface TabLineProps {
+  disabledKeys: Iterable<Key> | undefined,
+  isDisabled: boolean | undefined,
+  selectedTab: HTMLElement | undefined,
+  orientation?: Orientation
+}
+
+const selectedIndicator = style({
+  position: 'absolute',
+  color: {
+    default: 'neutral',
+    isDisabled: 'disabled',
+    forcedColors: {
+      default: 'Highlight',
+      isDisabled: 'GrayText'
+    }
+  },
+  height: size(2),
```

Comment:
Shouldn't change with scale or anything
```suggestion
  height: '[2px]',
```